dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerCommandBuilder.cs

```diff
@@ -515,11 +578,11 @@ private static SqlCommand CreateCommand(this SqlConnection connection, StringBui
         return command;
     }
 
-    private static void AddParameter(this SqlCommand command, VectorStoreRecordPropertyModel property, string name, object? value)
+    private static void AddParameter(this SqlCommand command, VectorStoreRecordPropertyModel? property, string name, object? value)
     {
         switch (value)
         {
-            case null when property.Type == typeof(byte[]):
+            case null when property is not null && property.Type == typeof(byte[]):
```

Comment:
Does this also work?
```suggestion
            case null when property?.Type is byte[]:
```

Reply:
Yeah, unfortunately not... But you can still do `when property?.Type == typeof(byte[])`.